### PR TITLE
Use furo

### PR DESCRIPTION
**Pull request by joaander**
## Description

Switch from `sphinx_rtd_theme` to `furo`.

## Motivation and context

Furo is a modern sphinx documentation style: https://pradyunsg.me/furo/
I like a lot of its features, including the dark mode, the clean styling, the right navigation bar, and the "Hide search matches" button. I have a number of problems with `sphinx_rtd_theme` which I have fixed over time with custom hacks in the CSS file deleted by this pull request. Going forward, it will be easier to maintain documentation when we make minimal customizations to it.

## How has this been tested?

I rendered the documentation locally. Reviewers can click the `readthedocs` link below to view it online.

## Change log

```
Changed:

* Use `furo` style for HTML documentation.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
Direct link to the docs for review: https://hoomd-blue--1498.org.readthedocs.build/en/1498/

File: sphinx-doc/_templates/page.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{% extends "furo/page.html" %}
+{% block footer %}
+<div class="related-pages">
+{% if next -%}
```
Comment (tommy-waltmann) [inline sphinx-doc/_templates/page.html:4]: What is this file for?

**Review by tommy-waltmann — COMMENTED** [review]
Overall, I like this new theme. My only comment on the documentation layout is that it would be nice if the entire python API wasn't shown as you scroll down the main page, maybe we could just have links to the main package and outer submodule hoomd, hpmc, and md.

File: sphinx-doc/_templates/page.html (hunk 1 above)
Comment (joaander) [reply sphinx-doc/_templates/page.html:4]: This is a Jinja template that replaces the `footer` block of furo's page https://github.com/pradyunsg/furo/blob/main/src/furo/theme/furo/page.html. I use it to add the UMich logo and the statement: "Development of HOOMD-blue is led by the Glotzer Group at the University of Michigan (supported by NSF DMR 1808342) with many external contributions." on every page of the documentation. This was done in `footer.html` for `sphinx-rtd-theme`.

**Comment by joaander** [comment]
> Overall, I like this new theme. My only comment on the documentation layout is that it would be nice if the entire python API wasn't shown as you scroll down the main page, maybe we could just have links to the main package and outer submodule hoomd, hpmc, and md.

OK, I made this change. The list has shown tree with a depth of 3 in the sphinx-rtd-theme documentation for quite some time.

**Comment by tommy-waltmann** [comment]
> > Overall, I like this new theme. My only comment on the documentation layout is that it would be nice if the entire python API wasn't shown as you scroll down the main page, maybe we could just have links to the main package and outer submodule hoomd, hpmc, and md.
> 
> OK, I made this change. The list has shown tree with a depth of 3 in the sphinx-rtd-theme documentation for quite some time.

Wow, I didn't realize this when I made the original comment. Either way, I think the look is better now.

**Review by tommy-waltmann — APPROVED** [review]
This is going to look really nice in our next release!

**Comment by joaander** [comment]
Thanks for taking the time to review and making suggestions to improve the documentation.

**Pull request merged**